app/src/main/java/com/github/polypoly/app/menu/JoinGameLobbyActivity.kt

```diff
@@ -491,73 +499,61 @@ class JoinGameLobbyActivity : MenuActivity("Join Game") {
     }
 
     /**
-     * This function is called when the user clicks on the button to join a gameLobby.
+     * This function changes the warning state according to the game lobby code input by the user
      * If the gameLobby code is empty, or the code is not in the DB, or the gameLobby is full,
      * it displays a warning message.
      * Otherwise, it calls the function to join the gameLobby.
      * @param warningState (MutableState<String>): The state of the warning message
      * @param mContext (Context): The context of the activity
-     * @return (String): The warning message to be displayed
      */
+    @SuppressLint("NewApi")
     private fun gameLobbyCodeButtonOnClick(warningState: MutableState<String>, mContext: Context) {
-        return if (gameLobbyCode.isEmpty()) {
+        if (gameLobbyCode.isEmpty()) {
             warningState.value = getString(R.string.game_lobby_code_is_empty)
-        } else if (!dbContainsGameLobbyCode(gameLobbyCode)) {
-            warningState.value = getString(R.string.game_lobby_does_not_exist)
-        } else if(gameLobbyIsFull(gameLobbyCode)){
-            warningState.value = getString(R.string.game_lobby_is_full)
         } else {
-            warningState.value = ""
-            joinGameLobbyRoom(mContext)
+            val lobbyKey = DB_GAME_LOBIES_PATH + gameLobbyCode
+            remoteDB.keyExists(lobbyKey).thenCompose { keyExists ->
+                if (!keyExists) {
+                    warningState.value = getString(R.string.game_lobby_does_not_exist)
+                    CompletableFuture.failedFuture(IllegalAccessException())
+                } else {
+                    remoteDB.getValue<GameLobby>(lobbyKey)
+                }
+            }.thenAccept { gameLobby ->
+                if(gameLobbyIsFull(gameLobby)){
+                    warningState.value = getString(R.string.game_lobby_is_full)
+                } else {
+                    warningState.value = ""
+                    joinGameLobbyRoom()
+                }
+            }
         }
     }
 
     /**
      * This function launches the gameLobby room activity and passes the gameLobby code to it.
      * @param mContext (Context): The context of the activity
      */
-    private fun joinGameLobbyRoom(mContext: Context) {
+    private fun joinGameLobbyRoom() {
         val currentLobbyKey = DB_GAME_LOBIES_PATH + gameLobbyCode
-        val gameLobby =  mockDb.getValue<GameLobby>(currentLobbyKey).get()
-        val newGameLobby = GameLobby(gameLobby.admin, gameLobby.gameMode, gameLobby.minimumNumberOfPlayers, gameLobby.maximumNumberOfPlayers, gameLobby.roundDuration
-            , gameLobby.gameMap, gameLobby.initialPlayerBalance, gameLobby.name, gameLobby.code, gameLobby.private)
-
-        for (player in gameLobby.usersRegistered) {
-            if (player != newGameLobby.admin) {
-                newGameLobby.addUser(player)
+        remoteDB.getValue<GameLobby>(currentLobbyKey).thenAccept { gameLobby ->
```

Comment:
As I said in #77, isn't it a bit too much? I mean you could do something like:
`if(!lobby.users.contains(currentUser)) then lobby.addUser(currentUser)`

Reply:
Nice one! I didn't even check that the user is not contained as otherwise the user would already be in the game.